modules/REST/V1/Settings.php

```diff
@@ -103,10 +103,21 @@ public function update_item( $request ) {
 		);
 
 		foreach ( $options as $option_name => $new_value ) {
-			$result = $this->options->set( $option_name, $new_value );
+			$previous_value = $this->options->get( $option_name );
+			$result         = $this->options->set( $option_name, $new_value );
 
 			if ( $result->has_errors() ) {
 				$errors->merge_from( $result );
+				continue;
+			}
+
+			if ( $this->options->get( $option_name ) === $previous_value ) {
+				continue;
+			}
```

Comment:
I suggest to avoid one call to `get()` and prevent useless `set()` if the new value is not different.
```php
			$previous_value = $this->options->get( $option_name );
			if ( $new_value === $previous_value ) {
				continue;
			}

			$result = $this->options->set( $option_name, $new_value );

			if ( $result->has_errors() ) {
				$errors->merge_from( $result );
				continue;
			}
```

Reply:
In this case the sanitized value can be different from the raw value. Example: all the boolean values: `1` => `true`. So the comparison must be done after `set()`.

Reply:
I primarily does nothing regarding the previous value and perform the update anyway.
Your commentaries confirm what I think, it complicates things.
Also, WP does the insertion anyway in `WP_REST_Settings_Controller::update_item()` so we may want to do the same here.

Regarding the client, if it send update request with the a value already set I think that performance issues is on its side.

If we really want to go that way, then we need proper error management to tell the client no update has been performed but not blocking error is raised either.
Otherwise it could do the request over and over without any changes.